### PR TITLE
Various fixes and improvements for molten blocks

### PR DESCRIPTION
**Pull request by MarioSMB**
- Breaking molten blocks with silk touch does not produce lava, breaking them without silk touch returns the regular block instead of its molten variant
- Support for Create's splashing mechanic to turn molten blocks into obsidian
- Molten block variants can be crafted by combining the block with a lava bucket
- Support for casting lava into blocks to obtain their molten variant in a Tinker's Construct smeltery, and melting them to obtain lava
- Molten block variants no longer function as replacements in vanilla recipes

This addresses most of the points in #40, besides hurting players walking on molten blocks (a little beyond my expertise currently). Happy to make any adjustments as needed.

### CODE REVIEW

File: src/main/resources/data/subwild/loot_tables/blocks/molten_andesite.json [hunk 1]
```diff
@@ -1,22 +1,43 @@
 {
```
Comment (PaintNinja) [inline src/main/resources/data/subwild/loot_tables/blocks/molten_andesite.json:1]: Please fix the indentation of this so the git comparison works. If you're editing a file indented with tabs, please make your changes also use tab indentation.

**Comment by PaintNinja** [comment]
Thanks for your PR! Had a quick glance over it and it looks good. There's some things I need sorted out before I can merge:
- Indentation consistency (mentioned above).
- Changes involving deleting files need to be explained. "Molten block variants no longer function as replacements in vanilla recipes" and "Don't allow using molten variants in place of their regular blocks in crafting recipes. This solves a few issues, especially if the above suggestions are considered" is helpful info but what specific issues does it solve?
- Being able to obtain molten variants of blocks with Tinker's Construct is great, however I would also like to see Vanilla options so that people aren't required to use separate mods for certain features. Maybe a crafting recipe that combines a lava bucket and a block to turn into molten? Maybe smelt a block to turn it molten like you can to turn cobble into stone?

File: src/main/resources/data/subwild/loot_tables/blocks/molten_andesite.json (hunk 1 above)
Comment (MarioSMB) [reply src/main/resources/data/subwild/loot_tables/blocks/molten_andesite.json]: Apologies, my changes were made based on vanilla loot tables, it should be resolved now.

**Comment by MarioSMB** [comment]
Alrighty, I've made those adjustments, including a mention in the OP about molten blocks being able to be crafted with a block and a lava bucket! I chose this recipe as smelting stone is already taken by smooth stone (along with other mods making use of the other blocks).
As for the removals; that is mostly so they don't conflict with their own recipes (recycling issues come up when the same block can be used to craft itself), but also because these blocks more closely resemble magma blocks than regular stone (both in functionality and visuals), though they of course shouldn't make use of magma block recipes either as they are unique enough to stand on their own. This especially makes sense now that they drop vanilla blocks rather than their molten form when broken normally.

**Pull request merged**